### ISSUE TITLE
Configurable deprecation warnings

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#29740]
### Describe the proposed change(s).

Add a new option "deprecationWarnings" with enum values:

- "disabled"
- "dashboard" (default, what we currently have)
- "combined-issue" (create a separate issue with all warnings inside instead of embedding in the Dashboard)
- "issue-per-package" (create one issue per package instead of in the dashboard)

In the case of the issues, the user should be allowed to close the issue to ignore it and it doesn't come back. We should log an info message if it's ignored.

**Comment by HonkingGoose** [comment]
I see there's a refactor PR that removes old code related to `deprecationWarningIssue`:

- #29736

But in this issue you want to have these options:

> - "combined-issue" (create a separate issue with all warnings inside instead of embedding in the Dashboard)
> - "issue-per-package" (create one issue per package instead of in the dashboard)

Do you intend to have _new_ code for these features? I wonder if there's anything worth saving from the "throw out old code PR". :wink:

**Comment by rarkins** [comment]
Yes, I think it's better to start over - that code is really old, and we can copy/paste if we need
